Wait for pending requests before writing test fixtures

antischematic opened the pull request:
Closes #12

antischematic committed:
Wait for pending requests before writing test fixtures

antischematic commented:
This could use some integration tests but not sure how to do that.

Sam152 commented:
Thanks! I'll study this in a bit more detail, this problem actually exists in the existing integration tests, so it'll be a good test case to switch over the "wait" approach to see if this works: https://github.com/Sam152/cypress-replay/blob/1.0/integration-tests/cypress/e2e/test_recording.cy.ts#L27

antischematic committed:
Use Map to store incoming requests

Handles the case where new requests are initiated while waiting for other requests to complete. These will be processed in the same waitForRequests queue.

Sam152 committed:
Remove sleep and re-record the integration test

Sam152 committed:
Add a test and rename function

Sam152 committed:
Add note in the docs

Sam152 committed:
Rename method again

Sam152 commented:
I've pushed a few cosmetic changes and changes to the test cases. I've also unbundled #12, because I think that's worth considering in isolation. I'm not convinced on that change, but will put thoughts in the associated issue.

Sam152 committed:
Rename variable
